tests/e2e/gc_rwx_reclaim_policy.go

```diff
@@ -0,0 +1,518 @@
+		10. Verify CnsVolumeMetadata CRD is created
+		11. Verify health status of PVC
+		12. Verify volume is created on CNS by using CNSQuery API and also check metadata is pushed to CNS
+		13. Create Pod using PVC-2 created above
+		14. Verify Pod creation fails
+		15. Verify the error message returned on failure is correct
+		16. Delete PVC and PVs from GC
+		17. Verify if PVC and PVs are deleted in the SV cluster and GC
+		18. Verify CnsVolumeMetadata CRD is deleted
+		19. Check if the VolumeID is deleted from CNS by using CNSQuery API
+	*/
+	ginkgo.It("Verify ReadWriteOnce PVC is not usable as ReadWriteMany", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		var storageclasspvc *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var err error
+
+		ginkgo.By("CNS_TEST: Running for GC setup")
+		scParameters[svStorageClassName] = storagePolicyName
+		ginkgo.By("Creating a PVC")
+		storageclasspvc, pvclaim, err = createPVCAndStorageClass(client,
+			namespace, nil, scParameters, diskSize, nil, "", false, v1.ReadWriteOnce)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client,
+			[]*v1.PersistentVolumeClaim{pvclaim}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
+
+		volHandle := persistentvolumes[0].Spec.CSI.VolumeHandle
+		gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+		volumeID := getVolumeIDFromSupervisorCluster(volHandle)
+		gomega.Expect(volumeID).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, pvclaim.Namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// Creating label for PV
+		// PVC will use this label as Selector to find PV
+		staticPVLabels := make(map[string]string)
+
+		ginkgo.By("Creating the PV in guest cluster")
+		pv2 := getPersistentVolumeSpecForRWX(volHandle, v1.PersistentVolumeReclaimDelete,
+			staticPVLabels, "2Gi", "", v1.ReadWriteMany)
+		pv2, err = client.CoreV1().PersistentVolumes().Create(ctx, pv2, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Creating the PVC in guest cluster")
+		pvc2 := getPersistentVolumeClaimSpecForRWX(namespace, staticPVLabels, pv2.Name, "2Gi")
+		pvc2, err = client.CoreV1().PersistentVolumeClaims(namespace).Create(ctx, pvc2, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Wait for PV and PVC to Bind
```

Comment:
the test is missing the `CnsVolumeMetadata` verification and volume health checks

Reply:
verifyVolumeMetadataInCNS is added